docs/plugins.md

```diff
@@ -34,10 +34,24 @@ spec:
 
 ### Security
 
-No security context is enforced, but is strong encouraged you configure your container not to run as root.
-
-Plugins can run at an non-localhost address, but this is very strongly discouraged. Within a pod, network traffic cannot
-be observed.
+Plugins are a powerful way to augment Argo Workflows' capabilities. Plugins also introduce code which may contain 
+vulnerabilities.
+
+Before enabling plugins, consider:
+* Who should be able to install plugins? How will this be enforced (e.g. RBAC rules for creating/modifying ConfigMaps)?
+* What should be the plugins' configured security context? Using non-root is strongly encouraged.
```

Comment:
The controller creates the agent pods, right? How will security contexts be configured?